source/elements/oneMKL/source/domains/data_fitting/interpolate.rst

```diff
@@ -0,0 +1,106 @@
+.. _interpolate:
+
+Interpolate Function
+====================
+
+.. contents::
+    :local:
+    :depth: 1
+
+Interpolate function performs computations of function and derivatives values at interpolation sites.
+
+If the sites do not belong to interpolation interval ``[a, b]``, the library uses:
+
+  - interpolant :math:`I_0` coefficients computed for interval :math:`[x_0, x_1)` for the
+    computations at the sites to the left of ``a``.
+  - interpolant :math:`I_{n-2}` coefficients computed for interval
+    :math:`[x_{n-2}, x_{n-1})` for the computations at the sites to the right of ``b``.
+
+Interpolation algorithm depends on interpolant's type (e.g., for cubic spline
+interpoilation evaluation of third-order polynomial is performed to obtain function values).
+
+Header File
+-----------
+
+.. code:: cpp
+
+  #include<oneapi/mkl/experimental/data_fitting.hpp>
+
+Namespace
+---------
+
+.. code:: cpp
+
+  oneapi::mkl::experimental::data_fitiing
+
+Syntax
+------
+
+.. code:: cpp
+
+  template <typename Interpolant>
+  sycl::event interpolate(
+      Interpolant& interpolant,
+      typename Interpolant::fp_type* sites,
+      std::int64_t n_sites,
+      typename Interpolant::fp_type* results,
+      const std::vector<sycl::event>& dependencies,
+      interpolate_hint ResultHint = interpolate_hint::funcs_sites_ders,
+      site_hint SiteHint = site_hint::non_uniform); // (1)
+
+  template <typename Interpolant>
+  sycl::event interpolate(
+      Interpolant& interpolant,
+      typename Interpolant::fp_type* sites,
+      std::int64_t n_sites,
+      typename Interpolant::fp_type* results,
+      std::bitset<32> der_indicator,
```

Comment:
I suppose so, but if I were you, I would choose a general pattern so it is predictable for all names...   lower snake case with full words ...  I see you sometimes shortening words sometimes not shortening them...  is there a pattern to it?  maybe you wan to rethink all the names into a single pattern :)

See if you can write down a general naming pattern:  recall this spec is supposed to be written so that another developer could implement the library, so you want to describe naming patterns when applicable in case someone wants to suggest an extension.

Reply:
on the other hand, I am ok with just a few shortened cases as long as it is readable... but you should review all your names and see if there is a general pattern that you can (and want to) write down... you may discover you want to be more consistent, or that you are ok with it :)  To me this is just one aspect of the full review.

Reply:
Ok. I see. Yeah, we need to think about general pattern and exceptions for it (if any). As for me interpolation sites will look too long... However, we definitely need to think about it. Thanks